### ISSUE TITLE
Unable to edit contribution pages with extension enabled

### ISSUE THREAD

**Issue by netzih** [progressivetech/net.ourpowerbase.remoteform#11]
Trying to edit contribution pages and cannot edit data.
When going to title (trying to configure page) I get an error x
Network Error
Unable to reach the server. Please refresh this page in your browser and try again.
All other pages I can edit (amount, profiles, campaign etc.).

**Comment by jmcclelland** [comment]
It sounds like there is an error trying to display the part of the form that allows you to enable remoteforms. Is there any information in the ConfigAndLog file? Or, can you enable the [web console](https://developer.mozilla.org/en-US/docs/Tools/Web_Console) and see if there are any helpful error messages?

**Comment by netzih** [comment]
I'm getting error 500 on the contribution management page

**Comment by netzih** [comment]
Is there maybe some setting on civi side I need to enable (besides enabling the extension itself)?

**Comment by jmcclelland** [comment]
That may be the case - I can't be sure. Does your web server error log give any indication of why the 500 error is being generated? If you see the error in the console, can you click on the Server Response area to try to see if it can tell you what the error is?

**Comment by netzih** [comment]
When using wordpress debug I'm getting
"Fatal error: Uncaught Error: Using $this when not in object context in /var/www/html/wordpress/wp-content/uploads/civicrm/ext/net.ourpowerbase.remoteform-master/remoteform.php:263 Stack trace: #0 /var/www/html/wordpress/wp-content/plugins/civicrm/civicrm/CRM/Utils/Hook.php(274): remoteform_civicrm_buildForm('CRM_Contribute_...', Object(CRM_Contribute_Form_ContributionPage_Settings)) #1 /var/www/html/wordpress/wp-content/plugins/civicrm/civicrm/CRM/Utils/Hook/WordPress.php(155): CRM_Utils_Hook->runHooks(Array, 'civicrm_buildFo...', 2, 'CRM_Contribute_...', Object(CRM_Contribute_Form_ContributionPage_Settings), NULL, NULL, NULL, NULL) #2 /var/www/html/wordpress/wp-content/plugins/civicrm/civicrm/Civi/Core/CiviEventDispatcher.php(90): CRM_Utils_Hook_WordPress->invokeViaUF(2, 'CRM_Contribute_...', Object(CRM_Contribute_Form_ContributionPage_Settings), NULL, NULL, NULL, NULL, 'civicrm_buildFo...') #3 /var/www/html/wordpress/wp-content/plugins/civicrm/civicrm/vendor/symfony/event-dispatcher/EventDispatcher.php(184): Civi\Core\C in /var/www/html/wordpress/wp-content/uploads/civicrm/ext/net.ourpowerbase.remoteform-master/remoteform.php on line 263"

Hope this can help

**Comment by jmcclelland** [comment]
Thank you! This is perfect - just what I needed. I just pushed a fix to this repository. If you update you should get it working again. I suspect different versions of php are more or less tolerant of the error I made - so it slipped by my testing, but you got stuck with it.

**Comment by netzih** [comment]
Awesome. this is now fixed although when putting it in remote website I am still getting the same error as I am with profiles. it is saying I added the URL's to the remote forms settings and enabled remote forms in the profile.
"Access to XMLHttpRequest at 'https://www.example.com/wp-admin/admin.php?page=CiviCRM&amp;q=civicrm%2Fremoteform' from origin 'https://www.remoteexample.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: Redirect is not allowed for a preflight request."
So still not getting it to display the form.
I have added the URL (put the www and without the www even though we only use www) but nothing is working.
